sphinx/locale/__init__.py

```diff
@@ -203,7 +203,12 @@ def setup(app):
     .. versionadded:: 1.8
     """
     def gettext(message: str) -> str:
-        return _TranslationProxy(catalog, namespace, message)  # type: ignore[return-value]
```

Comment:
This seemed like a nice cleanup -- and unlocks other simplifications -- so if it's possible to re-apply later in a separate change, then I'd like to do that.

Reply:
Which bit were you talking about here? Open to keeping it if my partial revert didn't.

A

Reply:
In particular: switching everything over to lazy-loading (so that the `gettext` method doesn't require any conditional logic) - and as a side-effect, allowing content to be localised into more than one language during a unit test.

Referring back to some previous [notes](https://github.com/sphinx-doc/sphinx/pull/10949#issue-1429065601):
> Attempting to write a test case to build the same application in two different languages was not initially possible -- the first-loaded translation catalog (as found in the sphinx.locale.translators global variable) -- would remain in-use for subsequent application builds under different locales

This can wait until after 6.2.x, unless you're particularly keen to dig into it.  I'd plan to take another look at it fresh in a few days' time.

Reply:
In particular: switching everything over to lazy-loading (so that the `gettext` method doesn't require any conditional logic) - and as a side-effect, allowing content to be localised into more than one language during a unit test.

Referring back to some previous [notes](https://github.com/sphinx-doc/sphinx/pull/10949#issue-1429065601):
> Attempting to write a test case to build the same application in two different languages was not initially possible -- the first-loaded translation catalog (as found in the sphinx.locale.translators global variable) -- would remain in-use for subsequent application builds under different locales

This can wait until after 6.2.x, unless you're particularly keen to dig into it.  I'd plan to take another look at it fresh in a few days' time.